ui/appui-react/src/appui-react/widgets/WidgetDef.tsx

```diff
@@ -578,3 +579,44 @@ export class WidgetDef {
     });
   }
 }
+
+/** @internal */
+export function getWidgetState(widgetDef: WidgetDef, state: NineZoneState) {
```

Comment:
Is there a reason you want this to be exported as a separate method? It seems that it fits perfectly inside `WidgetDef` class since we need to pass an object of that type anyway.

Reply:
With #766 this way we might keep the API a bit cleaner. Long term our strategy should be aimed at deprecating these classes..